p2p/gossip/protocol.go

```diff
@@ -245,6 +250,27 @@ func (prot *Protocol) isOldMessage(h hash) bool {
 	return oldmessage
 }
 
+func (prot *Protocol) processMessage(h hash, msg *pb.ProtocolMessage) error {
+	prot.markMessage(h)
+
+	var data service.Data
+
+	if payload := msg.GetPayload(); payload != nil {
+		data = service.DataBytes{Payload: payload}
+	} else if wrap := msg.GetMsg(); wrap != nil {
+		data = &service.DataMsgWrapper{Req: wrap.Req, MsgType: wrap.Type, ReqID: wrap.ReqID, Payload: wrap.Payload}
```

Comment:
I know this code was here before, but a wrapped message should come only when were having request response communication. so gossip should practically never get it. but maybe validate just in case?

Reply:
well, a protocol might choose to use the req-resp framework over Gossip - PoET might work this way for security reasons. I added a warning message in case the message is wrapped so that it will pop in the log and we can validate that it was used properly